docs/apis/subsystems/output.md

```diff
@@ -0,0 +1,281 @@
+---
+title: Output API
+tags:
+  - Output
+  - API
+---
+
+<!-- cspell:ignore renderables, sometext, courseid, enabletrusttext -->
+
+## Overview
+
+The Output API is responsible for visual aspects of Moodle content. This page explain how renderers, renderables, themes and templates all work together.
+
+The following documentation is also related:
+
+- The [Templates](../../guides/templates) used for rendering the output such as HTML or email messages bodies
+
+## Page Output Journey
+
+Lets start with building a page that is part of an admin tool.
+
+```php title="/admin/tool/demo/index.php"
+<?php
+// Standard GPL and phpdocs
+require_once(__DIR__ . '/../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+
+admin_externalpage_setup('tooldemo');
+
+// Set up the page.
+$title = get_string('pluginname', 'tool_demo');
+$pagetitle = $title;
+$url = new moodle_url("/admin/tool/demo/index.php");
+$PAGE->set_url($url);
+$PAGE->set_title($title);
+$PAGE->set_heading($title);
+
+$output = $PAGE->get_renderer('tool_demo');
+
+echo $output->header();
+echo $output->heading($pagetitle);
+
+$renderable = new \tool_demo\output\index_page('Some text');
+echo $output->render($renderable);
+
+echo $output->footer();
+
+```
+
+Explaining this demo step by step we have firstly the standard config and includes, plus the `admin_externalpage_setup('tooldemo')` which calls `require_login` and performs permissions checks for admin pages.
+
+```php
+require_once(__DIR__ . '/../../../config.php');
+require_once($CFG->libdir.'/adminlib.php');
+
+admin_externalpage_setup('tooldemo');
+```
+
+Then we load the title of the page from a lang string (see [String API](https://docs.moodle.org/dev/String_API) ). We use this string to set some `$PAGE` properties (URL title and heading).
+
+```php
+// Set up the page.
+$title = get_string('pluginname', 'tool_demo');
+$pagetitle = $title;
+$url = new moodle_url("/admin/tool/demo/index.php");
+$PAGE->set_url($url);
+$PAGE->set_title($title);
+$PAGE->set_heading($title);
+```
+
+:::note What is $PAGE and where did it come from ?
+
+`$PAGE` is a global variable used to track the state of the page that is being returned. It is an instance of the `moodle_page` class defined in `lib/pagelib.php`. See [Page API](https://docs.moodle.org/dev/Page_API) for more information on the `$PAGE` variable.
+
+:::
```

Comment:
This is largely irrelevant to the Output API docs.

I'd suggest putting it into a collapsed `<details>` section or an `:::info` section, for example:

```suggestion
:::info Setup of an admin page

On admin pages the `admin_externalpage_setup($sectionname);` function should be used to setup and perform permission checks, for example:

```php title="admin/tool/demo/mypage.php"
require_once(__DIR__ . '/../../config.php');
require_once("{$CFG->libdir}/adminlib.php");

admin_externalpage_setup('example');
```

:::

```

I'd suggest detailing the PAGE API still, but with more info.